puppet/modules/vault_client/templates/cert.service.erb

```diff
@@ -11,10 +11,10 @@ ExecStart=/opt/bin/vault-helper cert \
   <%= @common_name %> \
   <%= @base_path %> \
 <% if @alt_names.length > 0 -%>
-  '--san-hosts=<%= @alt_names.join(',') %>' \
+  --san-hosts=<%= @alt_names.join(',') %> \
```

Comment:
The quotes are not the problem here:

I guess you need to take a look at the vault-helper's role:

```Mar 08 08:24:17 ip-10-99-38-118.eu-west-1.compute.internal systemd[1]: Starting certificate service kubelet...
Mar 08 08:24:17 ip-10-99-38-118.eu-west-1.compute.internal vault-helper[7215]: time="2018-03-08T08:24:17Z" level=info msg="Renewed token: f259bb41-092d-c46e-896b-13139526cdc5"
Mar 08 08:24:17 ip-10-99-38-118.eu-west-1.compute.internal vault-helper[7215]: time="2018-03-08T08:24:17Z" level=info msg="Generating new certificates"
Mar 08 08:24:17 ip-10-99-38-118.eu-west-1.compute.internal vault-helper[7215]: time="2018-03-08T08:24:17Z" level=fatal msg="error requesting certificate: error writing CSR to vault at 'csire-cluster/pki/k8s/sign/kubelet': Error making API request.
Mar 08 08:24:17 ip-10-99-38-118.eu-west-1.compute.internal vault-helper[7215]: URL: PUT https://vault.tarmak.local:8200/v1/csire-cluster/pki/k8s/sign/kubelet
Mar 08 08:24:17 ip-10-99-38-118.eu-west-1.compute.internal vault-helper[7215]: Code: 400. Errors:
Mar 08 08:24:17 ip-10-99-38-118.eu-west-1.compute.internal vault-helper[7215]: * subject alternate name ip-10-99-38-118.eu-west-1.compute.internal not allowed by this role"
```